Fixed bug that closed text fields immediately after creating them with the middle mouse button.

Jared234 opened the pull request:
This is an attempt to fix the bug mentioned in #3983. After testing and debugging I realized that this bug not only appears if you use the middle mouse button, but also if you choose the right mouse button to open a text field.
When this function is used for other tools, releasing the pressed key should return to the previous tool (e.g. the eraser should disappear after releasing the middle/right mouse button). This is not the case with the text tool, since you not only want to create a text box, but also write something down right away.
Therefore, I decided to actually (permanently) select the text tool if you press the middle mouse button. If you think there is a better fix, jut let me know and I can have another look at it.

Jared234 commented:
@Technius Do you have any updates on whether this pull request can be merged into Master?

Technius reviewed (commented):
* Can you rebase onto `release-1.1` branch and change the merge target? This will help us get the bug fix into the next release, and we will also be able to merge the bugfix back into master eventually.
* I think a similar fix can be applied to the stylus tool as well (`StylusInputHandler.cpp`).

Jared234 commented:
Thanks for the review. I've added a fix for the stylus and rebased to the release 1.1 branch. I'm relatively new to Github, so if my rebase is wrong, just let me know and I'll try to fix it.

Technius commented:
I tested this locally and it works well.

I don't think you rebased correctly. What you did was merge `release-1.1` back into `master`, but we want to push the commits on to branch `release-1.1`.
1. First, you should drop the commits that are not your original commits using `git rebase -i master`. [See this page for a graphical explanation of rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing). Or an easier alternative is to copy the changes you made, run `git reset --hard master`, paste your changes, and then commit your changes again.
2. Run `git rebase release-1.1` to rebase on to `release-1.1` branch. If you run `git log`, you should then see that one of the commits in the log will be marked as `release-1.1` -- this means the rebase is correct.
3. Click edit next to the PR title: 
![image](https://user-images.githubusercontent.com/1066652/198897416-9794b8d9-23ce-47da-af73-23e076ea1f7a.png)

4. Choose `release-1.1` from the dropdown: 
![image](https://user-images.githubusercontent.com/1066652/198897440-d5ecd332-8988-4f1b-98f7-81be1fb5f166.png)

If you did this correctly, you should see only the commits you added for this PR -- and not a bunch of random commits from master.

Technius commented:
If you don't have time to rebase, I can handle the rebasing for you as well.

rolandlo commented:
> 2\. Run `git rebase release-1.1` to rebase on to `release-1.1` branch. If you run `git log`, you should then see that one of the commits in the log will be marked as `release-1.1` -- this means the rebase is correct.

Actually this works better than `git rebase release-1.1`:
```term
git rebase --onto release-1.1 master BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch, `3983_bug_text_tool_closing` in this case.
With `git rebase release-1.1` you will have to fix a lot of merge conflicts in my experience.

Jared234 commented:
I hope it works now. Thanks for the detailed instructions. 
I really appreciate your patience and your help.

Technius commented:
Thanks for the PR!
